### ISSUE TITLE
Symbol.for inconsistent return type warning

### ISSUE THREAD

**Issue by rx80** [google/closure-compiler#3422]
Test case, compiled with -O ADVANCED
```
/**
 * @return {symbol}
 */
function test() {
        return Symbol.for("test");
}
```
Outputs warning:
```
test.js:6: WARNING - [JSC_TYPE_MISMATCH] inconsistent return type
found   : (symbol|undefined)
required: symbol
        return Symbol.for("test");
               ^^^^^^^^^^^^^^^^^^
```
Expected: no warning, since Symbol.for always returns a symbol

**Comment by tjgq** [comment]
This was fixed in 8b42100. It will be on the next compiler release.
